src/libOpenImageIO/maketexture.cpp

```diff
@@ -1184,6 +1184,91 @@ make_texture_impl(ImageBufAlgo::MakeTextureMode mode, const ImageBuf* input,
+        // ACM SIGGRAPH / Eurographics Symposium on High-Performance Graphics 2018,
+        //
+        // Benedikt Bitterli
+        // https://benedikt-bitterli.me/histogram-tiling/
+
+
+        const float sigma   = configspec.get_float_attribute("maketx:cdfsigma");
+        const int bits      = configspec.get_int_attribute("maketx:cdfbits");
+        const uint64_t bins = 1 << bits;
+
+        // Normalization coefficient for the truncated normal distribution
+        const float c_sigma_inv = fast_erf(1.0f / (2.0f * M_SQRT2 * sigma));
+
+        // If there are channels other than R,G,B,A, we probably shouldn't do anything
+        // to them.
+
+        const int channels = std::min(4, src->spec().nchannels);
+
+        std::vector<float> invCDF(bins);
+        std::vector<float> CDF(bins);
+        std::vector<imagesize_t> hist;
+
+        for (int i = 0; i < channels; i++) {
+            hist = ImageBufAlgo::histogram(*src, i, bins, 0.0f, 1.0f);
+
+            // Turn the histogram into a non-normalized CDF
+            for (int j = 1; j < bins; j++) {
+                hist[j] += hist[j - 1];
+            }
+
+            // Store the inverse CDF as a lookup-table which we'll use to transform
+            // the image data to a Guassian distribution. As mentioned in Burley [2019]
+            // we're combining two steps here when using the invCDF lookup
+            // table: we first "look up" the image value through its CDF (the normalized histogram)
+            // which gives us a uniformly distributed value, which we're then feeding in to the
+            // Gaussian inverse CDF to transfrom the unifrom distribution to Gaussian.
+
+            for (int j = 0; j < bins; j++) {
+                float u = float(hist[j]) / hist[bins - 1];
+                float g = 0.5
+                          + sigma * M_SQRT2
+                                * fast_ierf(c_sigma_inv * (2 * u - 1));
+                invCDF[j] = std::min(1.0f, std::max(0.0f, g));
+            }
+            configspec.attribute("invCDF_" + std::to_string(i),
+                                 TypeDesc("float[" + std::to_string(bins) + "]"),
+                                 invCDF.data());
+
+
+            // Store the forward CDF as a lookup table to transform back
+            // to the original image distribution from a Gaussian distribution.
+
+            std::vector<float>::iterator upper;
+            for (int j = 0; j < bins; j++) {
+                upper  = std::upper_bound(invCDF.begin(), invCDF.end(),
+                                         float(j) / (float(bins - 1)));
+                CDF[j] = std::min(1.0f,
+                                  std::max(0.0f, float(upper - invCDF.begin())
+                                                     / float(bins - 1)));
+            }
+
+            configspec.attribute("CDF_" + std::to_string(i),
+                                 TypeDesc("float[" + std::to_string(bins) + "]"),
```

Comment:
I think this is more easily expressed as

    TypeDesc(TypeDesc::FLOAT, bins)

rather than constructing a string in multiple steps and then having to re-parse that string in the TypeDesc constructor.

Reply:
that does look a lot nicer, yes :)